code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -1410,11 +1410,12 @@
 	burst(M,P,damage_type)
 
 /datum/ammo/xeno/acid/on_hit_mob(mob/M, obj/item/projectile/P)
-	if(iscarbon(M))
-		var/mob/living/carbon/C = M
-		if(C.status_flags & XENO_HOST && istype(C.buckled, /obj/structure/bed/nest) || C.stat == DEAD)
-			return
-	..()
+	if(!iscarbon(M) || xeno_hivenumber(M) && xeno_hivenumber(M) == xeno_hivenumber(P.firer))
+		return ..()
```

Comment:
Are you sure you wanna do this? Shouldn't you just return here? Since there's no difference currently between this case and the mob being carbon or with a different hivenumber.

Reply:
Yea, actually it could just return without the ..() since the parent's is null.